compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/type/builders/BallerinaTupleTypeBuilder.java

```diff
@@ -68,16 +70,25 @@ public TypeBuilder.TUPLE withRestType(TypeSymbol restType) {
 
     @Override
     public TupleTypeSymbol build() {
-        List<BType> memberBTypes = new ArrayList<>();
-        for (TypeSymbol memberType : memberTypes) {
-            memberBTypes.add(getMemberType(memberType));
+        List<BTupleMember> memberTypes = new ArrayList<>();
+        for (TypeSymbol memberType : this.memberTypes) {
+            BType type = getMemberType(memberType);
+            BVarSymbol varSymbol;
+            if (type == symTable.noType) {
+                varSymbol = new BVarSymbol(0, null, null, symTable.noType, null,
+                        symTable.builtinPos, SymbolOrigin.VIRTUAL);
+            } else {
+                varSymbol = new BVarSymbol(type.flags, type.tsymbol.name, type.tsymbol.pkgID, type, type.tsymbol.owner,
+                        type.tsymbol.pos, type.tsymbol.origin);
```

Comment:
Can we extract this symbol creation out to a method and reuse wherever it's needed for a tuple member?